### PR TITLE
Treat IP Address as an optional parameter

### PR DESCRIPTION
**Pull request by ctyano**
# Description

IP addresses in X.509 Subject Alternative Names is an optional attribute in Athenz X.509 identity certificate.
Users shall set an empty string to POD_IP environment variable to remove IP address from a CSR.
If the IP address is not set in the CSR, ZTS will skip verifying the client IP address to match the SANs IP. (e.g. [InstanceZTSProvider](https://github.com/AthenZ/athenz/blob/7190de6e44253eb9186cf43d5f48060904c9ea39/libs/java/instance_provider/src/main/java/com/yahoo/athenz/instance/provider/impl/InstanceZTSProvider.java#L301-L304))

## Assignees
- [x] `Assignees` is set

## Type of changes
- [x] Apply one or more `labels` of the following that fits:
  - `bug`: Bug fix
  - `dependencies`: Dependency upgrades
  - `documentation`: Documentation changes
  - `enhancement`: New Feature
  - `good first issue`: First contribution
  - `logging`: Log changes
  - `refactor`: Refactoring (no functional changes, no api changes)

### Flags
- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code
---

## Checklist
- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

**Comment by mlajkim** [comment]
Please modify the `athenz-sia.env` file too.

- [ ] Modify: https://github.com/AthenZ/k8s-athenz-sia/blob/main/athenz-sia.env#L119

**Comment by mlajkim** [comment]
@ctyano Do you think we should merge this before the release of v3.2.0?

**Comment by mlajkim** [comment]
This requires operation check just in case in k8s platform.
TODOs:
- [ ] Operation done by @mlajkim

**Comment by mlajkim** [comment]
This won't be included in v3.2.0 release.

File: pkg/config/default.go [hunk 1]
```diff
@@ -86,7 +85,7 @@ func DefaultIdentityConfig() *IdentityConfig {
 		AthenzSuffix:              "",
 		ServiceAccount:            "",
 		SaTokenFile:               "",
-		PodIP:                     net.ParseIP("127.0.0.1"),
+		PodIP:                     nil,
```
Comment (mlajkim) [inline pkg/config/default.go:88]:
```suggestion
		PodIP:                     "",
```

Comment (t4niwa) [reply pkg/config/default.go:88]: @mlajkim `PodIP`が `net.IP`型なので`nil`でよいかなと思ったのですがいかがですかね

File: pkg/config/config.go [hunk 2]
```diff
@@ -118,10 +118,8 @@ func (idCfg *IdentityConfig) loadFromENV() error {
 
 	// parse values
 	var err error
-	idCfg.PodIP = net.ParseIP(idCfg.rawPodIP)
-	if idCfg.PodIP == nil {
-		// PodIP should always be non-nil to issue role certificate
-		return fmt.Errorf("Invalid POD_IP [%q]", idCfg.rawPodIP)
+	if idCfg.rawPodIP != "" {
+		idCfg.PodIP = net.ParseIP(idCfg.rawPodIP)
```
Comment (t4niwa) [inline pkg/config/config.go:122]: PodIPのparseに失敗した場合は、errorが発生する仕様になってると思いますので、以下のような追記をしたほうがよいかなと思いましたがいかがでしょうか。
```suggestion
		idCfg.PodIP = net.ParseIP(idCfg.rawPodIP)
		if idCfg.PodIP == nil {
			return fmt.Errorf("Invalid POD_IP [%q], %w", idCfg.rawPodIP, err)
		}
```

**Review by t4niwa — APPROVED** [review]

**Pull request merged**